saleor/site/models.py

```diff
@@ -159,3 +164,56 @@ def get_translated_keys(self):
             "header_text": self.header_text,
             "description": self.description,
         }
+
+
+class SiteByIdLoader(DataLoader):
+    context_key = "site_by_id"
+
+    def batch_load(self, keys):
+        sites_mapped = Site.objects.using(self.database_connection_name).in_bulk(keys)
+        return [sites_mapped.get(site_id) for site_id in keys]
+
+
+class SiteByHostLoader(DataLoader):
+    context_key = "site_by_host"
+
+    def batch_load(self, keys):
+        sites_mapped = Site.objects.using(self.database_connection_name).in_bulk(keys)
+        # simulate nonexisting `domain__iexact__in`
+        q_list = map(lambda k: Q(domain__iexact=k), keys)
+        q_list = reduce(lambda a, b: a | b, q_list)
+        sites = Site.objects.using(self.database_connection_name).filter(q_list)
+        sites_mapped = {s.domain.lower(): s for s in sites}
+        return [sites_mapped.get(host.lower()) for host in keys]
+
+
+def load_current_site(request):
+    from django.conf import settings
+
+    if getattr(settings, "SITE_ID", ""):
+        site_id = settings.SITE_ID
+        return SiteByIdLoader(request).load(site_id).get()
+
+    host = request.get_host()
+    site = SiteByHostLoader(request).load(host).get()
+
+    if site is None:
+        domain, port = split_domain_port(host)
+        site = SiteByHostLoader(request).load(domain).get()
+
+    if site is None:
+        raise ImproperlyConfigured(
+            'You\'re using the Django "sites framework" without having '
+            "set the SITE_ID setting. Create a site in your database and "
+            "set the SITE_ID setting."
+        )
+    else:
+        # Populate the other loader for free
+        SiteByIdLoader(request).prime(site.id, site)
+    return site
+
+
+def load_site(request, site_id=None):
```

Comment:
Do we need `site_id` here? AFAIR we always fetch site model by default `env.SITE_ID`. if not needed then we could simplify the loader, to only use `load_current_site`.

Reply:
@dekoza What about this comment? I don't see any usage of `load_site(info.context, site_id=<something>`. In that case, why do we need to have this?

Reply:
I removed it